omnigibson/systems/macro_particle_system.py

```diff
@@ -542,9 +613,15 @@ def _compute_batch_particles_position_orientation(cls, particles, local=False):
             link_tfs_batch = np.zeros((n_particles, 4, 4))
             particle_local_poses_batch = np.zeros_like(link_tfs_batch)
             for i, name in enumerate(particles):
-                link = cls._particles_info[name]["link"]
-                if link not in link_tfs:
-                    link_tfs[link] = T.pose2mat(link.get_position_orientation())
+                obj, link = cls._particles_info[name]["obj"], cls._particles_info[name]["link"]
+                if link is None:
+                    if obj not in link_tfs:
+                        # We want local object pose, NOT the physx pose -- since this will grab the root link
+                        # pose which we don't want!
+                        link_tfs[obj] = T.pose2mat(XFormPrim.get_local_pose(obj))
```

Comment:
sorry I am not sure I fully understand this. happy to discuss in person.

Reply:
Add more comments.

Reply:
Assert obj is directly under `/World/object`.

Reply:
done when particles are created

Reply:
added more comments